scripts/tank_cmd.py

```diff
@@ -176,11 +202,55 @@ def show_help(log):
 Launch maya for a folder:
 > tank /studio/proj_xyz/shots/ABC123 launch_maya
 
+Log out of the current user (no context required):
+> tank logout
+
 """
     for x in info.split("\n"):
         log.info(x)
 
 
+def ensure_authenticated(cmd_line_credentials):
+    """
+    Make sure that there is a current toolkit user set.
+    May prompt for a login/password if needed. Note that if command line
+    arguments are used, the current user will be overriden only for this
+    invocation of the tank command and future invocations without credentials
+    on the command line will use the current default user. If no command line
+    credentials are provided and no script user is set, then the user will be
+    prompted for his credentials and those will be remembered in future
+    invocations until that user logs out.
+
+    :param cmd_line_credentials: Dictionary of credentials passed from the
```

Comment:
That's because that's what _extract_credentials returns, which in and of itself is not necessarily a good reason. Also, there's also always two or the four parameters that are going to be None anyway.